### PR TITLE
libav: remove /bin/sh dependencies

### PR DESCRIPTION
**Pull request by xeji**
###### Motivation for this change

impurity resulted in a Hydra build fail on 18.03.

/cc ZHF #36453  

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
cc @shlevy please review

**Comment by dtzWill** [comment]
`bash` almost certainly should be a `nativeBuildInput`

**Comment by xeji** [comment]
One of the outputs installed in bin/ is a shell script, so I considered `bash` a runtime dependency.

**Comment by Mic92** [comment]
@GrahamcOfBorg build libav

File: pkgs/development/libraries/libav/default.nix [hunk 1]
```diff
@@ -45,7 +45,11 @@ let
       ++ optional (vpxSupport && hasPrefix "0.8." version) ./vpxenc-0.8.17-libvpx-1.5.patch
       ;
 
-    preConfigure = "patchShebangs doc/texi2pod.pl";
+    postPatch = ''
+      patchShebangs .
```
Comment (shlevy) [inline pkgs/development/libraries/libav/default.nix]: @Ericson2314 FYI another point that needs patchBuildShebangs

**Review by shlevy — APPROVED** [review]
This looks good to me. We will eventually have a notion of `patchBuildtimeShebangs` that will be more appropriate in `postPatch`, but since it doesn't exist yet this will just fail on cross. We can fix that later.

**Comment by Mic92** [comment]
@shlevy do you know what the motivation behind `/bin/sh` == busybox was?

**Comment by 7c6f434c** [comment]
@Mic92 well, a statically-linked busybox could help to solve #1424 completely.

**Comment by shlevy** [comment]
@Mic92 I think the main motivation is specifically static linking... bringing in a binary and all of its dependent libraries is sometimes a pain and sometimes impossible to do right, as @7c6f434c points out.

And to be honest, if it were up to me there'd be no /bin/sh in the sandbox at all (or it would just immediately error out with a "nice" error message). Many years ago I tried to argue that `system(3)` should take `sh` from PATH...

**Comment by Mic92** [comment]
@shlevy yes taking `/bin/sh` away would cause less issues then the current approach.

**Comment by Mic92** [comment]
@shlevy could we at least agree on something like `dash` then we would share a common ground with ubuntu/debian..

**Comment by shlevy** [comment]
My answer is always going to be in favor of things failing with /bin/sh, sorry :smile: I see the short-term value in a sane common base (though why `dash` instead of a fully POSIX `busybox` I don't see), but in the long term /bin/sh dependencies always bite us one way or another. Just don't depend on /bin/sh!

**Pull request merged**